osu.Game/Overlays/BeatmapSet/BeatmapPicker.cs

```diff
@@ -50,11 +52,11 @@ public BeatmapSetInfo BeatmapSet
 
         private void updateDisplay()
         {
-            difficulties.Clear();
+            Difficulties.Clear();
 
             if (BeatmapSet != null)
             {
-                difficulties.ChildrenEnumerable = BeatmapSet.Beatmaps.OrderBy(beatmap => beatmap.StarDifficulty).Select(b => new DifficultySelectorButton(b)
+                Difficulties.ChildrenEnumerable = BeatmapSet.Beatmaps.FindAll(b => b.Ruleset.Equals(ruleset.Value)).OrderBy(b => b.StarDifficulty).Select(b => new DifficultySelectorButton(b)
```

Comment:
Any reason this is not a `.Where()`? Could save a list allocation.

Reply:
Right.. I actually thought list functions are faster than linq, guess I should check the perfomance..